### PR TITLE
Feature/ios solution to kmm

### PR DESCRIPTION
**Pull request by yOoZZyOzzbourne**
Connected ios part to KMM/KMP monorepo

- Look mainly into `shared/rocket/build.gradle`

### CODE REVIEW

File: shared/rocket/build.gradle.kts [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+    }
+}
+
+val syncIosLibDebug by tasks.creating(Copy::class) {
+    dependsOn("assembleSharedDebugXCFramework")
+    from(layout.buildDirectory.dir("XCFrameworks/debug"))
+    into(layout.buildDirectory.dir(iosLibFolder))
+}
+
+val syncIosLibRelease by tasks.creating(Copy::class) {
```
Comment (yOoZZyOzzbourne) [inline shared/rocket/build.gradle.kts]: this @alesrazym

File: shared/rocket/build.gradle.kts [hunk 2]
```diff
@@ -1,19 +1,54 @@
+import org.jetbrains.kotlin.gradle.plugin.mpp.apple.XCFramework
+
 plugins {
     alias(libs.plugins.quanti.kmp.library)
     alias(libs.plugins.quanti.android.detekt)
     alias(libs.plugins.quanti.android.ktlint)
     alias(libs.plugins.kover)
+    alias(libs.plugins.kotlin.multiplatform)
```
Comment (alesrazym) [inline shared/rocket/build.gradle.kts]: This is extra, already applied within convention plugin `build-logic/convention/src/main/kotlin/plugins/KmpLibConventionPlugin.kt`

File: shared/rocket/build.gradle.kts [hunk 3]
```diff
@@ -1,19 +1,54 @@
...
 }
 
 android {
     namespace = "cz.quanti.rocketrepository"
 }
 
+val libName = "Shared"
+val iosLibFolder = "XCFrameworks/lib"
+
 multiplatformSwiftPackage {
```
Comment (alesrazym) [inline shared/rocket/build.gradle.kts]: so this is old way, not supporting simulator?

File: shared/rocket/build.gradle.kts [hunk 4]
```diff
@@ -1,19 +1,54 @@
...
-    packageName("RocketRepository")
+    packageName("Shared")
     swiftToolsVersion("5.3")
     targetPlatforms {
         iOS { v("13") }
     }
     outputDirectory(File(rootDir, "/"))
 }
+
+kotlin {
```
Comment (alesrazym) [inline shared/rocket/build.gradle.kts:17]: And this is the new way? I'd like to see it reusable, i.e. similar to `multiplatformSwiftPackage {}` configuration. Does this make sense? I can try to configure, as it's imo out of scope of your task :)

File: iOS/RocketApp.xcodeproj/xcshareddata/xcschemes/RocketApp.xcscheme [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+   version = "1.7">
+   <BuildAction
+      parallelizeBuildables = "YES"
+      buildImplicitDependencies = "YES">
+      <PreActions>
+         <ExecutionAction
+            ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
+            <ActionContent
+               title = "BuildKMM"
+               scriptText = "cd &quot;$PROJECT_DIR&quot;&#10;./scripts/build_xcframework_script.sh DEBUG&#10;">
```
Comment (alesrazym) [inline iOS/RocketApp.xcodeproj/xcshareddata/xcschemes/RocketApp.xcscheme:13]: hardcoded DEBUG. As I'm not familiar with Xcode development, is there a way to configure depending on `buildConfiguration = "Debug"` or similar?

File: iOS/RocketApp/Domain/Package.swift [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      dependencies: [
+        .product(name: "CoreToolkit", package: "Infrastructure"),
+        .product(name: "UIToolkit", package: "Infrastructure"),
+        .product(name: "NetworkClientExtensions", package: "Infrastructure"),
+        .product(name: "ErrorReporting", package: "swift-core"),
+        .product(name: "ModelConvertible", package: "swift-core"),
+        .product(name: "Networking", package: "swift-core"),
+        .product(name: "RequestBuilder", package: "swift-core"),
+        .product(name: "Dependencies", package: "swift-dependencies"),
+        .product(name: "fetchRockets", package: "mobile-assignment-kmm"),
```
Comment (alesrazym) [inline iOS/RocketApp/Domain/Package.swift:38]: this is the old way?

File: iOS/RocketApp/Domain/Package.swift [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        .product(name: "KMPNativeCoroutinesCore", package: "KMP-NativeCoroutines"),
+        .product(name: "KMPNativeCoroutinesAsync", package: "KMP-NativeCoroutines"),
+        .product(name: "Shared", package: "SDK")
```
Comment (alesrazym) [inline iOS/RocketApp/Domain/Package.swift:41]: and this is the new way?

**Review by alesrazym — COMMENTED** [review]
In general, it seems to be good idea to make it work on simulators. I think I understand the idea of connecting KMP module to Xcode project. Let's confirm my questions and discuss next steps.

File: iOS/RocketApp/Domain/Package.swift (hunk 6 above)
Comment (yOoZZyOzzbourne) [reply iOS/RocketApp/Domain/Package.swift:38]: yep

File: iOS/RocketApp/Domain/Package.swift (hunk 7 above)
Comment (yOoZZyOzzbourne) [reply iOS/RocketApp/Domain/Package.swift:41]: yes

File: shared/rocket/build.gradle.kts (hunk 3 above)
Comment (yOoZZyOzzbourne) [reply shared/rocket/build.gradle.kts]: No the old way is integrated into the project as a Package directly through dependency. Thus the KMP module is standalone library on git

File: iOS/RocketApp.xcodeproj/xcshareddata/xcschemes/RocketApp.xcscheme (hunk 5 above)
Comment (yOoZZyOzzbourne) [reply iOS/RocketApp.xcodeproj/xcshareddata/xcschemes/RocketApp.xcscheme:13]: Xcode has schemes that you can set for different development types - AdHoc, Release, Debug etc. Right now the only scheme we have on iOS is Debug. Due to absence of pipelines ( CI / CD ) its not that necesarry to add more schemes.

File: shared/rocket/build.gradle.kts (hunk 3 above)
Comment (alesrazym) [reply shared/rocket/build.gradle.kts]: try without this

**Review by alesrazym — APPROVED** [review]

**Pull request merged**